client/src/main/kotlin/no/nav/su/se/bakover/client/person/PersonClient.kt

```diff
@@ -97,4 +152,18 @@ internal class PersonClient(
             )
         }
     }
+
+    companion object {
+        internal fun <K, V> newCache(
+            maximumSize: Long = 500,
+            expireAfterWrite: Duration = Duration.ofMinutes(1),
+        ): Cache<K, V> {
+            return Caffeine.newBuilder().maximumSize(maximumSize)
+                // Merk at det ikke er noen garanti for at Caffeine rydder opp selvom en verdi er expired, slik at cachen potensielt kan ta stor plass.
+                // Les mer: https://github.com/ben-manes/caffeine/wiki/Cleanup
+                .expireAfterWrite(expireAfterWrite).build()
```

Comment:
Hva skjer hvis den blir full? Stopper å cache, eller tryner den? 

Tenker at 1 min sikkert er et greit sted å starte, selv om jeg også tror at vi like gjerne kunne cachet det i f.eks 1 time. Hva som er fornuftig avhenger vel i stor grad av bruksmønsteret (tanken bak "1 time") er vel at dette sannsynligvis dekker store deler av hele saksbehandlingen - "gevinst" i alle steg. Med 1 minutt mistenker jeg at vi bare får denne gevinsten for påfølgende kall innenfor samme kall (usikker på hvor mange av disse vi har), men at hvert nytt skjermbilde vil hente/cache på nytt. 

Litt spekulasjon fra min side der altså, kanskje du kunne lagt på noen metrikker for hit/miss i grafana? Hvis vi plutselig ender opp med 100% hit eller miss vet vi kanskje litt mer?

Reply:
Dersom den blir full, begynner den kaste ut de eldste. Vi bør følge litt med på dette sammen med størrelsen på cachen.

JWT-tokens varer bare 30 min før de fornyes, og vi bør gjøre nye oppslag når de fornyes.

Legger til noen metrikker for å ta en avgjørelse på hva vi gjør videre.

Reply:
Hvis den blir full, kastes den eldste ut.

JWT-tokenet varer ikke lengre enn 30 minutter, så utover det har ikke så mye hensikt slik som det er nå. Neste gang vi ser på det bør vi splitte opp i 2 caches. Som nevnt i annen kommentar.

Har lagt på metrikker som vi følger opp :)